internal/app/wwctl/node/add/main.go

```diff
@@ -22,17 +22,25 @@ func CobraRunE(vars *variables) func(cmd *cobra.Command, args []string) error {
 				return err
 			}
 		}
-		// remove the default network as all network values are assigned
+		// remove the UNDEF network as all network values are assigned
```

Comment:
Why is the behavior of NetDevs being changed as part of this commit? It doesn't seem to relate to adding structures for devices and filesystems.

Reply:
Right,  it doesn't relate directly.
I found a behavior for the NetDevs while adding the tests covering the recursive getter and setters. 
The tests before were aimed for the transformation from NodeConf to NodeYaml and back, the new tests are using the "real" `wwctl` commands, so the changes are combined here.